cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -1829,11 +1842,12 @@
 
 ;; Special-case 0.0 for floats to use the `(zero_reg)` directly.
 ;; See #7162 for why this doesn't fall out of the rules below.
-(rule 8 (imm $F32 0) (gen_bitcast (zero_reg) $I32 $F32))
-(rule 8 (imm $F64 0) (gen_bitcast (zero_reg) $I64 $F64))
+(rule 9 (imm (ty_supported_float $F16) 0) (gen_bitcast (zero_reg) $I16 $F16))
+(rule 9 (imm (ty_supported_float $F32) 0) (gen_bitcast (zero_reg) $I32 $F32))
+(rule 9 (imm (ty_supported_float $F64) 0) (gen_bitcast (zero_reg) $I64 $F64))
 
 ;; If Zfa is enabled, we can load certain constants with the `fli` instruction.
-(rule 7 (imm (ty_supported_float ty) imm)
+(rule 8 (imm (ty_supported_float ty) imm)
   (if-let $true (has_zfa))
   (if-let const (fli_constant_from_u64 ty imm))
```

Comment:
Right, I was hoping to avoid dealing with that in this initial version, so I'm going to restrict `fli` to 32/64 bits and come back to it in a future PR.